### ISSUE TITLE
[Currency] Workspace - After changing workspace currency, approving, then unapproving an expense the amount updates not the currency

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#44989]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.5-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create a workspace and enable work flows and enable approvals
3. Submit an expense in the workspace chat
4. Go to workspace settings and change the currency
5. Go to workspace chat and open the submitted expense report
6. Click on the approve button on the header and observe the system message
7. Click on down arrow to the right of the title then click `Unapprove`

## Expected Result:
Both the currency and the amount get updated to the correct one

## Actual Result:
The amount gets updated to reflect the correct amount after conversion, but the currency still remains the same

## Workaround:
Unnown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/686e0a36-0c9e-46e5-b361-bfc31e162676

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019897b64867377d98</li>
        <li>Upwork Job ID: 1813371632470305179</li>
        <li>Last Price Increase: 2024-10-16</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @hungvu193</details>

**Comment by lanitochka17** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by mallenexpensify** [comment]
<img width="515" alt="image" src="https://github.com/Expensify/App/assets/22508304/54075444-162e-4f13-b68a-c3d318b2c556">

Before the bug shows, this happens which I'm unsure about. The approved says ETB and the unapproved shows Br.

Earlier in the vid it shows `ETB - Br` as the default currency.  Need to figure out why it's showing like this and if they're considered the same.

**Comment by mallenexpensify** [comment]
App crashed on me when I tried to repro, posted about here
https://github.com/Expensify/App/issues/44410#issuecomment-2215861787

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by mallenexpensify** [comment]
@hungvu193 can you attempt reproduction? I don't think I was able to. 


https://github.com/user-attachments/assets/1f1d39da-04c5-4731-aad1-cb42e7206339

**Comment by hungvu193** [comment]
@mallenexpensify Same here. 
@lanitochka17 can you still reproduce this one?

**Comment by mallenexpensify** [comment]
I threw `retest-weekly`, if QA isn't able to repro we'll close.

**Comment by lanitochka17** [comment]
Issue still reproducible when unapproving but when approving issue does not get reproducible.


https://github.com/user-attachments/assets/8628a2c7-2666-46dc-aadc-921ba46c94b4

**Comment by mallenexpensify** [comment]
@hungvu193 can you please test based on the vid above? It looks like I was able to repro
<img width="280" alt="image" src="https://github.com/user-attachments/assets/26cdb0f2-6220-4413-b964-3d91110ef293">

Assuming you are, I'll update the steps in OP then open this up for proposals

**Comment by hungvu193** [comment]
Sure I'll give it a try today

**Comment by hungvu193** [comment]
Ok, I can reproduce this one

<img width="807" alt="Screenshot 2024-07-31 at 14 04 24" src="https://github.com/user-attachments/assets/345618b3-89d1-48a4-88b7-eae95e2d2454">

**Comment by hungvu193** [comment]
I think this is BE issue, API returned wrong amount after being unapproved

<img width="409" alt="Screenshot 2024-07-31 at 14 08 39" src="https://github.com/user-attachments/assets/15ea0fd9-bfc4-4ba7-bc10-3581f52024ec">

**Comment by mallenexpensify** [comment]
@hungvu193 can you add 🎀 to assign an internal engineer to confirm it's backend?  Thx

**Comment by hungvu193** [comment]
Sure.

🎀 👀 🎀

**Comment by hungvu193** [comment]
Not overdue, this is BE issue

**Comment by puneetlath** [comment]
I agree it's a back-end issue. Added the hot-picks label and assigned to wave-collect.

**Comment by trjExpensify** [comment]
Unapprove is built in #wave-control, moving. CC: @Beamanator @garrettmknight

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.

https://github.com/user-attachments/assets/87e039c2-888f-4676-b3e3-ebe82b10065a

**Comment by Beamanator** [comment]
I'm happy to take this on, but it would be great if we could clean this issue up a little bit 🙏 
1. The issue title still mentions `approving` but we're mainly talking about unapproving now, right?
2. Do we have super reproducible steps? We still have the label `Needs Reproduction` so that makes me think that we don't yet?

**Comment by mallenexpensify** [comment]
Gotcha @Beamanator , title and OP updated.  it was just this that needed to be added at the end of the steps
> 7. Click on down arrow to the right of the title then click `Unapprove`

I was able to reproduce just now
<img width="366" alt="image" src="https://github.com/user-attachments/assets/e8bd3562-4b11-4e67-aace-22e5cbf49961">

**Comment by mallenexpensify** [comment]
Testing something

**Comment by Beamanator** [comment]
ohhhh baby, i'll try to check this out today

**Comment by Beamanator** [comment]
planning to look this week

**Comment by Beamanator** [comment]
this weeeeek

**Comment by Beamanator** [comment]
still haven't had time :/

**Comment by Beamanator** [comment]
same, def this week

**Comment by Beamanator** [comment]
Ok sorry this took so long to get back to, but now that I'm trying to reproduce locally, I don't actually see the same bug happening, do you @mallenexpensify @hungvu193 ? If not, let's add `retest-weekly` to see if applause can reproduce. Here's what I'm seeing:

https://github.com/user-attachments/assets/ff1b6232-4f16-4f6c-a096-c4952371dbcc

**Comment by mallenexpensify** [comment]
Threw retest-weekly on it, bumped to weekly and made me owner. Thx for checking/testing Alex

**Comment by Beamanator** [comment]
Waiting for retest

**Comment by mallenexpensify** [comment]
Should be retested soon

**Comment by mallenexpensify** [comment]
Still waiting on retest

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.


https://github.com/user-attachments/assets/36680729-4a6b-4c4b-a5d1-f00ae07356a2

**Comment by mallenexpensify** [comment]
Adding `Help Wanted` back and bumping to $500 to get 👀. 
I tried to repro yesterday and ran into another issue :/ 
Will test again soon, if/when needed

**Comment by mallenexpensify** [comment]
Updated title to inc. `[Currency]` 
Waiting to get picked up internally, it's on #expense so not a near term priority

**Comment by trjExpensify** [comment]
Someone can run into this on a collect workspace, right? If so, I'd move this to #quality as a general bug.

**Comment by Beamanator** [comment]
haven't had a chance to look at this yet, hopefully later this week

**Comment by Beamanator** [comment]
Can we add a priority here so I know if I should be hardcore focusing this one?

**Comment by mallenexpensify** [comment]
I was looking at the rubric for #quality and it doesn't seem to easily fit there
<img width="303" alt="SnagitHelper2024 2024-11-19 13 31 58" src="https://github.com/user-attachments/assets/6c0a1d24-5f24-4f0e-bb44-e3b0359466be">

<img width="814" alt="image" src="https://github.com/user-attachments/assets/f5710e20-4894-4162-bdb2-c1cc25cfdfca">

@muttmuure do you think this should be #quality?  I see many other posts in the Slack room with `currency` `in:#quality currency`

**Comment by mallenexpensify** [comment]
@Beamanator , I think the priority here is MEDIUM because it's a bug that a real user _could_ experience and it's also related to _real_ money, where the expense would be reported incorrectly and the user wouldn't be repaid the correct amount.

**Comment by garrettmknight** [comment]
@mallenexpensify does this change the actual expense amount? I can't tell if it's a front end viewing issue or an actual back end change to the expense amount and that makes a big difference on the importance of fixing to me.

**Comment by twisterdotcom** [comment]
I am removing hot pick from any issues with an internal engineer assigned so other engineers can know it's available.

**Comment by garrettmknight** [comment]
This feels like it could affect our current cohort so I think we should fix it in a cohort room. @hungvu193 can you take a look and see if this is FE or if it needs an internal engineer?

**Comment by hungvu193** [comment]
Oops I missed this notification, I'll check this one tomorrow morning.

**Comment by mallenexpensify** [comment]
Bumping to `Daily` and making @hungvu193 owner, for now.

**Comment by tgolen** [comment]
I'm gonna grab this one and look into it.

**Comment by tgolen** [comment]
I was not able to reproduce this issue locally. In my testing, the workspace began in currency USD. I submitted a $5 expense in USD. After submitting the expense, I updated the workspace to have GBP as the default currency. Approving and unapproving the expense from the admin account leaves the expense always as a $5 USD expense.

The reproduction steps are not clear. When they say "the correct currency", I do not know what the "correct" one is supposed to be. Please rewrite the reproduction steps using actual currency and amount values so that I am aware of what exactly to expect if you are still able to reproduce this problem.

**Comment by mallenexpensify** [comment]
I tested and was unable to reproduce, please comment/reopen if you're able to reproduce and can provide reliable reproduction steps (and a video!)
